src/platform/forms-system/src/js/web-component-patterns/fullNamePattern.js

```diff
@@ -247,6 +247,14 @@ const firstNameLastNameNoSuffixDef = cloneDeep(fullNameNoSuffixSchema);
 delete firstNameLastNameNoSuffixDef.properties.middle;
 const firstNameLastNameNoSuffixSchema = firstNameLastNameNoSuffixDef;
 
+/**
```

Comment:
Instead of doing this and creating a new pattern I think it would be better to turn the `fullNameNoSuffixSchema` into a function that can accept options which will configure the character limits, which are now hard-coded. Maybe something like: 
```
const fullNameNoSuffixSchema = options => {
  const { maxLength = 30, minLength = 1 } = options || {};
  return {
    type: 'object',
    required: ['first', 'last'],
    properties: {
      first: {
        type: 'string',
        minLength,
        maxLength,
      },
      middle: {
        type: 'string',
        maxLength,
      },
      last: {
        type: 'string',
        minLength,
        maxLength,
      },
    },
  };
};
```

Reply:
Happened upon this (while working on VA webforms BIO) and apologize as I have nit read all the code, but the concept of NoMaxLength concerns me as that could open the door to DOS simply be feeding an infinite string into the field. I believe 256 (or 1024 if you want huge) should be good enough.

Reply:
Great point @openprivacy, I just implemented a 256 character limit.  Thanks